devlib/target.py

```diff
@@ -2226,14 +2226,16 @@ async def is_screen_on(self):
                 return True
             return boolean(match.group(1))
         else:
-            raise TargetStableError('Could not establish screen state.')
+            self.logger.debug('Could not get screen state (probably off?).')
```

Comment:
That's a bit dodgy. If we end up on that else branch, it does not mean the screen is off, it means we don't know, and the original code was reflecting that. I don't think it's a good idea to remove that and just blindly assume the screen is off. 

If client code does not care about distinction between screen off and screen unknown, we should make a new exception type (inheriting from TargetStableError) so we don't accidentally catch unrelated TargetStableError and then catch it in the code in question.

Reply:
Well, let's start with `ANDROID_SCREEN_STATE_REGEX` then. I'm unsure if `adb dumpsys power` is the correct way of probing screen state. Why not using `adb dumpsys window` if we are trying to determine whether the screen is on?
Maybe we should extend `is_screen_on()` to tell screen lock state (locked, unlocked), too. Probably it should be renamed (e.g., `get_screen_state()`) as well.

Reply:
I vaguely remember having issues with using `dumpsys window` to determine this state in the past and `dumpsys power` was the most reliable method at the time. It's possible that things have changed in later versions of Android so we could even have some hybrid to try and ensure we get a reliable reading rather than making any assumptions. 

Also I would suggest leaving `is_screen_on` as a standalone method as it is, but adding something else to determine the lock state sounds like it could be useful.

Reply:
`is_screen_on()` raises `Could not establish screen state` exception if my Pixel 6 (Android-12) is in idle state (screen is completely off) instead of `Asleep`. This is why I removed that exception.
I could handle the exception and try to wake up device (via keyevent 26), but this does not sound right solution, too.

Reply:
Added https://github.com/ARM-software/devlib/pull/659/commits/50424e3f7567892ca2246ab89db030aa4e6c2182 to handle `Could not establish screen state` exception when the device is idling.